internal/controllers/utils/utils.go

```diff
@@ -1112,10 +1125,8 @@ func ClusterIsReadyForPolicyConfig(
 }
 
 // TimeoutExceeded returns true if it's been more time than the timeout configuration.
-func TimeoutExceeded(provisioningRequest *provisioningv1alpha1.ProvisioningRequest) bool {
-	timeSince := time.Since(provisioningRequest.Status.ClusterDetails.NonCompliantAt.Time)
-	timeout := time.Duration(provisioningRequest.Spec.Timeout.Configuration) * time.Minute
-	return timeSince > timeout
+func TimeoutExceeded(startTime time.Time, timeout time.Duration) bool {
```

Comment:
I think it would be more clear to leave this function as it is and pass a timeout in minutes.
If not, every time this function is used, the caller will have to transform the timeout beforehand.